semantic-python/src/Language/Python/Core.hs

```diff
@@ -16,13 +18,35 @@ import qualified TreeSitter.Python.AST as Py
 class Compile py where
   -- FIXME: we should really try not to fail
   compile :: (Member Core sig, Carrier sig t, Foldable t, MonadFail m) => py -> m (t Name)
+
   default compile :: (MonadFail m, Show py) => py -> m (t Name)
   compile = defaultCompile
 
+  compileCC :: (Member Core sig, Carrier sig t, Foldable t, MonadFail m) => py -> m (t Name) -> m (t Name)
+
+  default compileCC :: ( Member Core sig
+                       , Carrier sig t
+                       , Foldable t
+                       , MonadFail m
+                       )
+                    => py -> m (t Name) -> m (t Name)
+  compileCC py cc = (>>>) <$> compile py <*> cc
+
+-- | TODO: This is not right, it should be a reference to a Preluded
+-- NoneType instance, but it will do for now.
+none :: (Member Core sig, Carrier sig t) => t Name
+none = unit
+
 defaultCompile :: (MonadFail m, Show py) => py -> m (t Name)
 defaultCompile t = fail $ "compilation unimplemented for " <> show t
 
-instance (Compile l, Compile r) => Compile (Either l r) where compile = compileSum
+newtype CompileSum py = CompileSum py
+
+instance (Generic py, GCompileSum (Rep py)) => Compile (CompileSum py) where
+  compile (CompileSum a) = compileSum a
+  compileCC (CompileSum a) cc = compileCCSum a cc
```

Comment:
What do you think about removing `compileSum` & `compileCCSum` so we can’t accidentally use one but not the other?